x/profiles/keeper/handler.go

```diff
@@ -156,6 +156,12 @@ func handleMsgDeleteProfile(ctx sdk.Context, keeper Keeper, msg types.MsgDeleteP
 
 // handleMsgRequestDTagTransfer handles the request of a dTag transfer
 func handleMsgRequestDTagTransfer(ctx sdk.Context, keeper Keeper, msg types.MsgRequestDTagTransfer) (*sdk.Result, error) {
+	// check if the request's receiver has blocked the sender before
+	if keeper.IsUserBlocked(ctx, msg.Receiver, msg.Sender) {
+		return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest,
+			fmt.Sprintf("The user with address %s has been blocked from %s", msg.Sender, msg.Receiver))
```

Comment:
I think that since it's the same use that's creating the reaction and that has been blocked, we can return a more specific error:

```suggestion
			fmt.Sprintf("The user with address %s has blocked you", msg.Sender))
```